src/libcore/tests/iter.rs

```diff
@@ -1427,9 +1427,9 @@ fn test_range_inclusive_nth() {
     assert_eq!(r, 13..=20);
     assert_eq!(r.nth(2), Some(15));
     assert_eq!(r, 16..=20);
-    assert_eq!(r.is_empty(), false);
+    assert_eq!(ExactSizeIterator::is_empty(&r), false);
     assert_eq!(r.nth(10), None);
-    assert_eq!(r.is_empty(), true);
+    assert_eq!(ExactSizeIterator::is_empty(&r), true);
```

Comment:
Is this going to be a concern for end users?

Reply:
> ExactSizeIterator::is_empty (also unstable)

ah, nvm

Reply:
`Range::is_empty` is strictly more general, so it's not a problem in a world where everything is stable.  It may be a slight hiccup in nightly, but I would expect it to be solved in the wild by adding the new feature gate, rather than the annoying UFCS.  Here I just changed it to the other method because I didn't want to change what the test was testing.  (It _would_ be a problem if we wanted to stabilize `ExactSizeIterator::is_empty` first, as mentioned in the OP, but I feel like this should be less controversial than the new trait method, especially since there are a bunch of iterators that aren't `ExactSize` that could provide an efficient `is_empty`.)